src/rust/ide/src/executor/test_utils.rs

```diff
@@ -61,12 +63,29 @@ impl TestWithLocalPoolExecutor {
             self.run_task(task);
         }
     }
+
+    /// Runs all tasks in the pool and returns if no more progress can be made on any task.
+    pub fn run_until_stalled(&mut self) {
+        self.executor.run_until_stalled();
+    }
```

Comment:
If we exposed API this way, perhaps we don't need `when_stalled` any more? Instead of:
```rust
let mut test = TestWithLocalPoolExecutor::set_up()
...
test.when_stalled(|| { ..something...});
```
We can have just
```rust
let mut test = TestWithLocalPoolExecutor::set_up()
...
test.run_until_stalled();
 ..something...
```

And if somehow it's still needed, it should use self.run_until_stalled instead of  self.executor.run_until_stalled();

Reply:
I think both APIs should stay in place. Well, the latter is more general, so we could rewrite all our tests on top of it but there's no value.
Basically `when_stalled` allows you to setup responses and then run the asynchronous tasks with `await` sugar and whatnot.
Without this, you need to to manually poll the futures. It is sometimes necessary (e.g. as you want to test that something completes only after receiving some external stimulus) but I'd rather do that only when justified by the scenario we want to test.

The latter comment will be applied.